### ISSUE TITLE
[Performance]: Inference triggers segmentation fault on multi-CPU Xeon with nthreads exceeding single CPU

### ISSUE THREAD

**Issue by maticomp** [openvinotoolkit/openvino#21354]
### OpenVINO Version

2023.2.0

### Operating System

Ubuntu 22.04 (LTS)

### Device used for inference

CPU

### OpenVINO installation

PyPi

### Programming Language

Python

### Hardware Architecture

x86 (64 bits)

### Model used

https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/intel/person-detection-0200/README.md

### Model quantization

Yes

### Target Platform

```
ubuntu@ip-172-31-24-184:~$ lscpu
Architecture:            x86_64
  CPU op-mode(s):        32-bit, 64-bit
  Address sizes:         46 bits physical, 48 bits virtual
  Byte Order:            Little Endian
CPU(s):                  96
  On-line CPU(s) list:   0-95
Vendor ID:               GenuineIntel
  Model name:            Intel(R) Xeon(R) Platinum 8275CL CPU @ 3.00GHz
    CPU family:          6
    Model:               85
    Thread(s) per core:  2
    Core(s) per socket:  24
    Socket(s):           2
    Stepping:            7
    CPU max MHz:         3900.0000
    CPU min MHz:         1200.0000
    BogoMIPS:            6000.00
    Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm const
                         ant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma
                         cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cd
                         p_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2
                          erms invpcid cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cq
                         m_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req pku ospke avx512_vnni md_clear flush_l1d arch_capab
                         ilities
Virtualization features:
  Virtualization:        VT-x
Caches (sum of all):
  L1d:                   1.5 MiB (48 instances)
  L1i:                   1.5 MiB (48 instances)
  L2:                    48 MiB (48 instances)
  L3:                    71.5 MiB (2 instances)
NUMA:
  NUMA node(s):          2
  NUMA node0 CPU(s):     0-23,48-71
  NUMA node1 CPU(s):     24-47,72-95
Vulnerabilities:
  Gather data sampling:  Mitigation; Microcode
  Itlb multihit:         KVM: Mitigation: VMX disabled
  L1tf:                  Not affected
  Mds:                   Not affected
  Meltdown:              Not affected
  Mmio stale data:       Mitigation; Clear CPU buffers; SMT vulnerable
  Retbleed:              Mitigation; Enhanced IBRS
  Spec store bypass:     Mitigation; Speculative Store Bypass disabled via prctl
  Spectre v1:            Mitigation; usercopy/swapgs barriers and __user pointer sanitization
  Spectre v2:            Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence
  Srbds:                 Not affected
  Tsx async abort:       Mitigation; TSX disabled
ubuntu@ip-172-31-24-184:~$ lscpu -e
CPU NODE SOCKET CORE L1d:L1i:L2:L3 ONLINE    MAXMHZ    MINMHZ      MHZ
  0    0      0    0 0:0:0:0          yes 3900.0000 1200.0000 3000.000
  1    0      0    1 1:1:1:0          yes 3900.0000 1200.0000 3000.000
  2    0      0    2 2:2:2:0          yes 3900.0000 1200.0000 3000.000
  3    0      0    3 3:3:3:0          yes 3900.0000 1200.0000 3000.000
  4    0      0    4 4:4:4:0          yes 3900.0000 1200.0000 3000.000
  5    0      0    5 5:5:5:0          yes 3900.0000 1200.0000 3000.000
  6    0      0    6 6:6:6:0          yes 3900.0000 1200.0000 3000.000
  7    0      0    7 8:8:8:0          yes 3900.0000 1200.0000 3000.000
  8    0      0    8 9:9:9:0          yes 3900.0000 1200.0000 3000.000
  9    0      0    9 10:10:10:0       yes 3900.0000 1200.0000 3000.000
 10    0      0   10 11:11:11:0       yes 3900.0000 1200.0000 3000.000
 11    0      0   11 12:12:12:0       yes 3900.0000 1200.0000 3000.000
 12    0      0   12 13:13:13:0       yes 3900.0000 1200.0000 3000.000
 13    0      0   13 16:16:16:0       yes 3900.0000 1200.0000 3000.000
 14    0      0   14 17:17:17:0       yes 3900.0000 1200.0000 3000.000
 15    0      0   15 18:18:18:0       yes 3900.0000 1200.0000 3000.000
 16    0      0   16 19:19:19:0       yes 3900.0000 1200.0000 3000.000
 17    0      0   17 20:20:20:0       yes 3900.0000 1200.0000 3000.000
 18    0      0   18 21:21:21:0       yes 3900.0000 1200.0000 3000.000
 19    0      0   19 25:25:25:0       yes 3900.0000 1200.0000 3000.000
 20    0      0   20 26:26:26:0       yes 3900.0000 1200.0000 3000.000
 21    0      0   21 27:27:27:0       yes 3900.0000 1200.0000 3000.000
 22    0      0   22 28:28:28:0       yes 3900.0000 1200.0000 3000.000
 23    0      0   23 29:29:29:0       yes 3900.0000 1200.0000 3000.000
 24    1      1   24 32:32:32:1       yes 3900.0000 1200.0000 3000.000
 25    1      1   25 33:33:33:1       yes 3900.0000 1200.0000 3000.000
 26    1      1   26 34:34:34:1       yes 3900.0000 1200.0000 3000.000
 27    1      1   27 35:35:35:1       yes 3900.0000 1200.0000 3000.000
 28    1      1   28 36:36:36:1       yes 3900.0000 1200.0000 3000.000
 29    1      1   29 37:37:37:1       yes 3900.0000 1200.0000 3000.000
 30    1      1   30 38:38:38:1       yes 3900.0000 1200.0000 3000.000
 31    1      1   31 40:40:40:1       yes 3900.0000 1200.0000 3000.000
 32    1      1   32 41:41:41:1       yes 3900.0000 1200.0000 3000.000
 33    1      1   33 42:42:42:1       yes 3900.0000 1200.0000 3000.000
 34    1      1   34 43:43:43:1       yes 3900.0000 1200.0000 3000.000
 35    1      1   35 44:44:44:1       yes 3900.0000 1200.0000 3000.000
 36    1      1   36 45:45:45:1       yes 3900.0000 1200.0000 3000.000
 37    1      1   37 48:48:48:1       yes 3900.0000 1200.0000 3000.000
 38    1      1   38 49:49:49:1       yes 3900.0000 1200.0000 3000.000
 39    1      1   39 50:50:50:1       yes 3900.0000 1200.0000 3000.000
 40    1      1   40 51:51:51:1       yes 3900.0000 1200.0000 3000.000
 41    1      1   41 52:52:52:1       yes 3900.0000 1200.0000 3000.000
 42    1      1   42 53:53:53:1       yes 3900.0000 1200.0000 3000.000
 43    1      1   43 57:57:57:1       yes 3900.0000 1200.0000 3000.000
 44    1      1   44 58:58:58:1       yes 3900.0000 1200.0000 3000.000
 45    1      1   45 59:59:59:1       yes 3900.0000 1200.0000 3000.000
 46    1      1   46 60:60:60:1       yes 3900.0000 1200.0000 3000.000
 47    1      1   47 61:61:61:1       yes 3900.0000 1200.0000 3000.000
 48    0      0    0 0:0:0:0          yes 3900.0000 1200.0000 1200.097
 49    0      0    1 1:1:1:0          yes 3900.0000 1200.0000 3000.000
 50    0      0    2 2:2:2:0          yes 3900.0000 1200.0000 3000.000
 51    0      0    3 3:3:3:0          yes 3900.0000 1200.0000 3000.000
 52    0      0    4 4:4:4:0          yes 3900.0000 1200.0000 3000.000
 53    0      0    5 5:5:5:0          yes 3900.0000 1200.0000 3000.000
 54    0      0    6 6:6:6:0          yes 3900.0000 1200.0000 3000.000
 55    0      0    7 8:8:8:0          yes 3900.0000 1200.0000 3000.000
 56    0      0    8 9:9:9:0          yes 3900.0000 1200.0000 3000.000
 57    0      0    9 10:10:10:0       yes 3900.0000 1200.0000 3000.000
 58    0      0   10 11:11:11:0       yes 3900.0000 1200.0000 3000.000
 59    0      0   11 12:12:12:0       yes 3900.0000 1200.0000 3000.000
 60    0      0   12 13:13:13:0       yes 3900.0000 1200.0000 3000.000
 61    0      0   13 16:16:16:0       yes 3900.0000 1200.0000 3000.000
 62    0      0   14 17:17:17:0       yes 3900.0000 1200.0000 3000.000
 63    0      0   15 18:18:18:0       yes 3900.0000 1200.0000 3000.000
 64    0      0   16 19:19:19:0       yes 3900.0000 1200.0000 3000.000
 65    0      0   17 20:20:20:0       yes 3900.0000 1200.0000 3000.000
 66    0      0   18 21:21:21:0       yes 3900.0000 1200.0000 3000.000
 67    0      0   19 25:25:25:0       yes 3900.0000 1200.0000 3000.000
 68    0      0   20 26:26:26:0       yes 3900.0000 1200.0000 3000.000
 69    0      0   21 27:27:27:0       yes 3900.0000 1200.0000 3000.000
 70    0      0   22 28:28:28:0       yes 3900.0000 1200.0000 3000.000
 71    0      0   23 29:29:29:0       yes 3900.0000 1200.0000 3000.000
 72    1      1   24 32:32:32:1       yes 3900.0000 1200.0000 3000.000
 73    1      1   25 33:33:33:1       yes 3900.0000 1200.0000 3000.000
 74    1      1   26 34:34:34:1       yes 3900.0000 1200.0000 3000.000
 75    1      1   27 35:35:35:1       yes 3900.0000 1200.0000 3000.000
 76    1      1   28 36:36:36:1       yes 3900.0000 1200.0000 3000.000
 77    1      1   29 37:37:37:1       yes 3900.0000 1200.0000 3000.000
 78    1      1   30 38:38:38:1       yes 3900.0000 1200.0000 3000.000
 79    1      1   31 40:40:40:1       yes 3900.0000 1200.0000 3000.000
 80    1      1   32 41:41:41:1       yes 3900.0000 1200.0000 3000.000
 81    1      1   33 42:42:42:1       yes 3900.0000 1200.0000 3000.000
 82    1      1   34 43:43:43:1       yes 3900.0000 1200.0000 3000.000
 83    1      1   35 44:44:44:1       yes 3900.0000 1200.0000 3000.000
 84    1      1   36 45:45:45:1       yes 3900.0000 1200.0000 3000.000
 85    1      1   37 48:48:48:1       yes 3900.0000 1200.0000 3000.000
 86    1      1   38 49:49:49:1       yes 3900.0000 1200.0000 3000.000
 87    1      1   39 50:50:50:1       yes 3900.0000 1200.0000 3000.000
 88    1      1   40 51:51:51:1       yes 3900.0000 1200.0000 3000.000
 89    1      1   41 52:52:52:1       yes 3900.0000 1200.0000 3000.000
 90    1      1   42 53:53:53:1       yes 3900.0000 1200.0000 3000.000
 91    1      1   43 57:57:57:1       yes 3900.0000 1200.0000 3000.000
 92    1      1   44 58:58:58:1       yes 3900.0000 1200.0000 3000.000
 93    1      1   45 59:59:59:1       yes 3900.0000 1200.0000 3000.000
 94    1      1   46 60:60:60:1       yes 3900.0000 1200.0000 3000.000
 95    1      1   47 61:61:61:1       yes 3900.0000 1200.0000 3000.000
ubuntu@ip-172-31-24-184:~$
```

### Performance issue description

Hello!

I'm trying to utilize the full CPU power of the Xeon-based server with 2 CPUs, and my inference attempts end up with a segmentation fault. I was able to reproduce this with `benchmark_app` and isolate this issue. I hope you might be able to provide some insights as to what I am missing in the puzzle.

I'm using the `person_detection_0200` model from the model zoo for benchmarking, but the issue happens on all models I tried, including my home-grown ones.

When running the default benchmark as:

```
benchmark_app -m person-detection-0200.xml
```

The benchmark runs fine, but utilizes only one half of the available logical cores:

![image](https://github.com/openvinotoolkit/openvino/assets/135164/4b3df70d-9b21-41a8-8e2c-9fdaadb0ae42)

```
[Step 8/11] Querying optimal runtime parameters
[ INFO ] Model:
[ INFO ]   NETWORK_NAME: torch-jit-export
[ INFO ]   OPTIMAL_NUMBER_OF_INFER_REQUESTS: 12
[ INFO ]   NUM_STREAMS: 12
[ INFO ]   AFFINITY: Affinity.CORE
[ INFO ]   INFERENCE_NUM_THREADS: 48
[ INFO ]   PERF_COUNT: False
[ INFO ]   INFERENCE_PRECISION_HINT: <Type: 'float32'>
[ INFO ]   PERFORMANCE_HINT: PerformanceMode.THROUGHPUT
[ INFO ]   EXECUTION_MODE_HINT: ExecutionMode.PERFORMANCE
[ INFO ]   PERFORMANCE_HINT_NUM_REQUESTS: 0
[ INFO ]   ENABLE_CPU_PINNING: True
[ INFO ]   SCHEDULING_CORE_TYPE: SchedulingCoreType.ANY_CORE
[ INFO ]   ENABLE_HYPER_THREADING: False
[ INFO ]   EXECUTION_DEVICES: ['CPU']
[ INFO ]   CPU_DENORMALS_OPTIMIZATION: False
[ INFO ]   CPU_SPARSE_WEIGHTS_DECOMPRESSION_RATE: 1.0
[Step 9/11] Creating infer requests and preparing input tensors
[ WARNING ] No input files were given for input 'image'!. This input will be filled with random values!
[ INFO ] Fill input 'image' with random values
[Step 10/11] Measuring performance (Start inference asynchronously, 12 inference requests, limits: 60000 ms duration)
[ INFO ] Benchmarking in inference only mode (inputs filling are not included in measurement loop).
[ INFO ] First inference took 5.25 ms
[Step 11/11] Dumping statistics report
[ INFO ] Execution Devices:['CPU']
[ INFO ] Count:            154404 iterations
[ INFO ] Duration:         60007.40 ms
[ INFO ] Latency:
[ INFO ]    Median:        4.46 ms
[ INFO ]    Average:       4.47 ms
[ INFO ]    Min:           4.17 ms
[ INFO ]    Max:           13.09 ms
[ INFO ] Throughput:   2573.08 FPS
```

I followed the discussion at https://github.com/openvinotoolkit/openvino/issues/15994 and I understand the default behaviour is intended, unless you provide number of threads for model compilation. So I did:

```
benchmark_app -nthreads 96 -hint none -m person-detection-0200.xml
```

But this ends up with a segmentation fault. 

```
[Step 6/11] Configuring input of the model
[ INFO ] Model inputs:
[ INFO ]     image (node: image) : u8 / [N,C,H,W] / [1,3,256,256]
[ INFO ] Model outputs:
[ INFO ]     detection_out (node: detection_out) : f32 / [...] / [1,1,200,7]
[Step 7/11] Loading the model to the device
Segmentation fault (core dumped)
```

Surprisingly, a smaller, but still higher than 48, number of threads does not segfault:

```
benchmark_app -nthreads 71 -hint none -m person-detection-0200.xml
```

But it doesn't utilize the full CPU either. 72 threads and onwards, it's a segmentation fault. The number is always half-way through the second CPU core number (so, 48+48/2 in this example). 

![image](https://github.com/openvinotoolkit/openvino/assets/135164/287d398d-0496-4c9c-a23c-1fcf17b82674)

Running two simultaneous copies of the `benchmark_app` doesn't spill to the other CPU either, saturating half of the cores and giving half of the throughput for both copies of the app.

What would be the logical next step to debug the segmentation fault, or to change the configuration to fully utilize the available resources?

The aforementioned results come from AWS's metal EC2, but I've been consistently reproducing this on physical devices where I first encountered this. The setup is always the same: clean-slate Ubuntu 22.04 with OpenVINO 2023.2 installed through APT from the official repository, and the `benchmark_app` coming from the distribution itself.

From the "Issue submission checklist" I understand I might have misclassified this as a performance issue, but I'm unsure if this is a bug in itself. The segmentation fault coming from the official benchmark application lead me to believe it's not an open question. If I am mistaken, my sincere apologies.

Thanks! 

### Issue submission checklist

- [X] I'm reporting a performance issue. It's not a question.
- [X] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [X] There is reproducer code and related data files such as images, videos, models, etc.

**Comment by maticomp** [comment]
An unexpected ripple effect of that is that OpenCV (built from source with openvino) triggers the same segmentation fault with the simple inference run on a model loaded via `readNetFromModelOptimizer`:

```
import cv2
import numpy as np

model = cv2.dnn.readNetFromModelOptimizer(
    "person-detection-0200.xml", "person-detection-0200.bin"
)

model.setInput(
    cv2.dnn.blobFromImage(
        cv2.cvtColor(np.zeros((256, 256, 3), np.uint8), cv2.COLOR_BGR2RGB),
        1.0,
        (256, 256),
    )
)

print(model.forward())
```

It will first spew an error and then result in a SIGSEGV:

```
ubuntu@ip-172-31-24-184:~$ python3 python_test.py
Traceback (most recent call last):
  File "/home/ubuntu/python_test.py", line 16, in <module>
    print(model.forward())
cv2.error: OpenCV(4.8.1) /home/ubuntu/opencv-python/opencv/modules/dnn/src/ie_ngraph.cpp:865: error: (-2:Unspecified error) in function 'initPlugin'
> Failed to initialize Inference Engine backend (device = CPU): Exception from src/inference/src/core.cpp:113:
> [ GENERAL_ERROR ] Unknown socket id -1
>
ubuntu@ip-172-31-24-184:~$ python3 python_test.py
Segmentation fault (core dumped)
ubuntu@ip-172-31-24-184:~$
```

**Comment by yuxu42** [comment]
@maticomp First of all, thanks for the issue! If I'm understanding your question correctly, you want to infer the model "person-detection-0200.xml" with OpenVINO to leverage all cpu core across the 2-scoket Xeon CPUs. In most of cases, from the performance perspective, it's the most performant since the cross socket inference means the cross-socket memory access which performance penalty. If you want get the best latency but also considering the total througput on the whole platform.
It's suggested to use the command below. To have two streams, and each stream running on CPUs within the single socket.
benchmark_app -hint latency -m person-detection-0200.xml
Or: 
benchmark_app -nthreads 96 -nstreams 2 -hint none -m person-detection-0200.xml

BTW, if you pursue the best overall throughput on the whole platform, you can use the command:
benchmark_app -hint throughput -m person-detection-0200.xml

**Comment by maticomp** [comment]
@yuxu42 yes, I ultimatealy want to leverage the full power of the machine with the 2 CPUs. 

But my issue came to light most importantly because my Python inference code was segfaulting on me for no apparent reason (see my first comment), which lead me to try to isolate the issue.

The command you suggest:

```
benchmark_app -nthreads 96 -nstreams 2 -hint none -m person-detection-0200.xml
```

indeed does not issue a segmentation fault, but still utilizes only half of the logical cores:

![image](https://github.com/openvinotoolkit/openvino/assets/135164/e011e289-cd12-47d2-b6a8-bb88a81e1815)

Running it with the `-hint latency` gives very similar result. In fact, I don't see much difference regardless of `nstreams` value in that approach (I tried small and big numbers). Removing the `-nstreams` param gets back to segmentation fault issue though.

So, in fact, how to utilize all 96 logical processors, while important, is somewhat a side question to this one:

> Why running `benchmark_app` with `-nthreads 96` *and* running a vanilla Python code using OpenCV for inference on this platform issues a segmentation fault instead of running properly.

To work around this, I'm now pinning the whole inference application in Python/OpenCV to a subset of cores by explicitly forcing affinity on logical cores, but this feels like fighting a problem that shouldn't be here.

Thanks!

**Comment by yuxu42** [comment]
@maticomp Thanks for the experiments and snapshot! From the snapshot, we can see all CPU physical cores are running on both sockets. Actually this is desired CPU core usage for #-hint latency to get the best performance. For Intel CPU ISA, each physical core and logical core pair is sharing the same accelator (SSE/AVX2/AVX512) for AI workload acceleration, so we do not use both physical core and logical core, but only physical cores.

@wangleis could you please comment on the CPU core usage? With the #-nthreads 96 args, only physical cores are used, but not both physical cores and logical cores.

**Comment by maticomp** [comment]
> For Intel CPU ISA, each physical core and logical core pair is sharing the same accelator

Okay, that makes a lot of sense. I understand that the 50% of logical cores utilization may be the desired behaviour. However this is not entirely consistent with my observations (see below).

Now, the question remains why adding more processing threads leads to an immediate segfault instead of just "not doing things any faster". And this, in practice, affects different inference code samples that do not explicitly set the number of threads, like the OpenCV one I listed above.

I redid the tests on a single CPU machine with 48 logical cores (c5.12xlarge instead of c5.24xlarge) on AWS.

The observations are as follows:

* `benchmark_app -m person-detection-0200.xml` saturates all 48 logical cores
* `benchmark_app -hint tput -m person-detection-0200.xml` saturates all 48 logical cores
* `benchmark_app -nthreads 24 -hint none -m person-detection-0200.xml` saturates half of the cores (24)
* `benchmark_app -nthreads 48 -hint none -m person-detection-0200.xml` saturates half of the cores (24), but *does not crash*
* `benchmark_app -nthreads 96 -hint none -m person-detection-0200.xml` saturates half of the cores (24), but *does not crash*

So the segmentation fault issue is isolated to multi-CPU systems, it seems.

**Comment by yuxu42** [comment]
It looks like there is the issue of single instance running on CPU core across socket as you mentioned. But this use case (single instance running across socket) will not give performance gain since the performace gain from more cpu cores on the other socket cannot balance the performance penalty of cross socket memory access as I mentioned previously.
With regards to best performance, we suggest simply use the performance hint. The logic inside OV RT analyzes the model and target platform, and then determine the appropriate thread#, stream#, CPU cores and etc to give the best performance. 
`benchmark_app -hint latency -m person-detection-0200.xml`
`benchmark_app -hint tput -m person-detection-0200.xml`

**Comment by maticomp** [comment]
Thank you @yuxu42. I understand I can control this behaviour explicitly with OV, and that doing the right thing with choosing the number of threads and performance hints is the way to yield the best performance.

But sorry about being so pushy about this, none of this explains the segmentation faults happening on multi-CPU machines, but not on single-CPU machines.

My core, practical issue is described in my first comment: https://github.com/openvinotoolkit/openvino/issues/21354#issuecomment-1830898957 - I'm running a piece of software that uses OpenCV with OpenVINO support to load the model, with no further way to provide OV hints (since it's not using Python's `openvino` module at all). While this works perfectly fine on a single-CPU hardware (and I _don't know_ what actual inference settings will be used under the hood), it simply crashes on a multi-CPU setup. I found no other way to resolve this than to pin the software explicilty to a given set of CPU cores.

What I believe is that OpenVINO should **not crash** _regardless_ of the `nthreads` settings used, thus I raised this issue.

I understand I may have subpar performance when choosing wrong approach, but I would expect this to _execute without crashing_.

**Comment by maticomp** [comment]
Thank you @wangleis! The first section is clear to me, thanks for explaining. I'm looking forward to hearing from you regarding the segmentation fault.

Shall I monitor this issue, or will another one be split from this one?

**Comment by xipingyan** [comment]
Root-cause to Core part, @wangleis is working on fixing this bug.

**Comment by maticomp** [comment]
@wangleis Unfortunately I can't quickly rebuild OpenCV with the new ov to test the Python scenario, but I built fresh ov alone from the master branch and all `benchmark_app` scenarios that were crashing now work flawlessly. Thank you! 

Is this expected to be included in 2023.3 when it's released?

**Comment by wangleis** [comment]
@maticomp Yes, the fix will be included in next release.
